x-pack/plugins/fleet/server/types/models/preconfiguration.ts

```diff
@@ -130,7 +130,7 @@ export const PreconfiguredFleetProxiesSchema = schema.arrayOf(
 export const PreconfiguredAgentPoliciesSchema = schema.arrayOf(
   schema.object({
     ...AgentPolicyBaseSchema,
-    namespace: schema.maybe(NamespaceSchema),
+    namespace: schema.maybe(AgentPolicyNamespaceSchema),
```

Comment:
shouldn't the agent policy namespace be required here?

Reply:
I think this will introduce a breaking change, but we should probably have a default value no?

Reply:
I'm not sure why there was a `schema.maybe` there. I think you right, I didn't notice it, I was basically enforcing it from the schema itself (has `lenght: 1` as requirement). I'll fix it

Reply:
@nchaulet that's true 🤦‍♀️ We could leave as it is, the AgentPolicySchema actually enforces it already:
https://github.com/elastic/kibana/pull/174776/files#diff-7c570055afd7a69dcc80854cb7559a0735b7580eab6ec1254e7344bea549a274R15-R23

To clarify, this is not doing any change on the current functionality, it's just separating the schema for agent policy namespace (that requires to have namespace) and package policy namespace (that can be optional). So I wouldn't do any further changes here.

Reply:
wouldn't it be enough to use the same `NamespaceSchema` with a `schema.maybe` wrapped around when using it in package policy?

Reply:
> wouldn't it be enough to use the same NamespaceSchema with a schema.maybe wrapped around when using it in package policy?

I had it like that and without the `minLength: 1` the agent policy schema allowed the field to be empty, that's why I redefined it. I don't think it is enough.